src/cpu_info.rs

```diff
@@ -0,0 +1,125 @@
+use slog::info;
+
+use crate::{OxidebpfError, LOGGER};
+
+pub fn online() -> Result<Vec<i32>, OxidebpfError> {
+    let cpu_string = std::fs::read_to_string("/sys/devices/system/cpu/online").map_err(|e| {
+        info!(
+            LOGGER.0,
+            "cpu_info::online(); could not read /sys/devices/system/cpu/online; error: {:?}", e
+        );
+        OxidebpfError::FileIOError
+    })?;
+
+    process_cpu_string(&cpu_string)
+}
+
+pub fn max_possible_index() -> Result<usize, OxidebpfError> {
+    let cpu_string = std::fs::read_to_string("/sys/devices/system/cpu/possible").map_err(
+        |e| {
+            info!(
+                LOGGER.0,
+                "cpu_info::max_possible_index(); could not read /sys/devices/system/cpu/possible; error: {:?}", e
+            );
+            OxidebpfError::FileIOError
+        },
+    )?;
+
+    max_index(&cpu_string)
```

Comment:
Anecdotally, the `online` and `possible` formats seem to be the same. In the average case, they'll match 1:1. In the edge case, it may be a superset but the highest number is not necessarily the highest count. For example, I can't see any reason `0-11,16-32` isn't a valid `possible`, where `online` might be `2-5,20-24`. Weird, but possible.

I can't find the place in the kernel where this string is written, but also looking at where CPU masks are set I don't see why this wouldn't be _possible_. Since we're already dealing with edge cases, let's just count explicitly here instead of taking the max index.